docs/glossary.md

```diff
@@ -0,0 +1,81 @@
+# Glossary of Terms
```

Comment:
Note: In the future, we should update the terms to include language-modeling-specific examples (e.g. mentioning the action space consists of next tokens)